raster/src/main/scala/geotrellis/raster/render/ColorMap.scala

```diff
@@ -228,8 +230,15 @@ class IntColorMap(breaksToColors: Map[Int, Int], val options: Options = Options.
 
   def cache(h: Histogram[Int]): ColorMap = {
     val ch = h.mutable
-    h.foreachValue(z => ch.setItem(z, map(z)))
-    new IntCachedColorMap(colors, ch, options)
+    val cachedColors = h.values()
+    cfor(0)( _ < cachedColors.length, _ + 1) { i =>
```

Comment:
Eh, nothing smart, that's why i approved it. I tried to avoid somehow calculating length for the second time (as it's already known in fact) and trying to avoid distinct function usage. But no; it's fine as it is.